src/test/kotlin/io/bazel/kotlin/integration/MultiplexTest.kt

```diff
@@ -0,0 +1,216 @@
+          out = "",
+          err = exception.toString()
+        )
+      }
+      .getOrThrow()
+  }
+
+  fun WriteWorkspace.Package.one(contents: WriteWorkspace.Package.() -> Unit): WriteWorkspace.Resolve {
+    return "library/one"{
+      apply(contents)
+    }
+  }
+
+  fun WriteWorkspace.Package.two(contents: WriteWorkspace.Package.() -> Unit): WriteWorkspace.Resolve {
+    return "library/two"{
+      apply(contents)
+    }
+  }
+
+  @Test
+  fun multipleBuilds() {
+    val workspace = WriteWorkspace.using<MultiplexTest> {
+      build {
+        val (define_kt_toolchain) = load("@$rulesRepo//kotlin:core.bzl", "define_kt_toolchain")
+        define_kt_toolchain(
+          "experimental_multiplex_workers" to True
+        )
+      }
+
+      val (location, archive_repository) = defineArchiveRepository()
+      workspace {
+        load(location, archive_repository)
+        archive_repository(
+          "name" to rulesRepo,
+          "path" to writeReleaseArchive().toString()
+        )
+        load("@$rulesRepo//kotlin:repositories.bzl", "kotlin_repositories")
+        "kotlin_repositories"()
+
+        load("@$rulesRepo//kotlin:core.bzl", "kt_register_toolchains")
+        "kt_register_toolchains"()
+      }
+
+      val lib = one {
+        build {
+          val ktJvmLibrary = loadKtJvmLibrary()
+
+          ktJvmLibrary(
+            "name" to "one",
+            "srcs" to !"glob(['*.kt'])",
+            "visibility" to !"['//:__subpackages__']"
+          )
+        }
+
+        kotlin("One.kt") {
+          +"package library.one"
+          "interface One" {
+            "fun one"(!"value:String")
+          }
+        }
+      }
+
+      two {
+        build {
+          val ktJvmLibrary = loadKtJvmLibrary()
+
+          ktJvmLibrary(
+            "name" to "two",
+            "srcs" to !"glob(['*.kt'])",
+            "deps" to !"[${lib.target("one")}]",
+            "visibility" to !"['//:__subpackages__']"
+          )
+        }
+        kotlin("Two.kt") {
+          +"package library.two"
+          +"import library.one.One"
+          "class Two : One" {
+            "override fun one"(!"value:String") {
+              "println"(!"value")
+            }
+          }
+        }
+      }
+    }
+
+    (0..50).forEach { round ->
+      workspace.build("//library/two").run {
+        assertWithMessage("Out:\n$out\nErr:\n$err").that(code).isEqualTo(0)
+        println("$round:\n$err\n\n")
+      }
+      WriteWorkspace.open(workspace) {
+        when (round % 3) {
+          0 -> {
+            one {
+              kotlin("One.kt") {}
+            }
+            two {
+              kotlin("Two.kt") {
+                +"package library.two"
+                "class Two" {
+                  "fun one"(!"value:String") {
+                    "println"(!"value")
+                  }
+                }
+              }
+            }
+          }
+          1 -> {
+            one {
+              kotlin("One.kt") {
```

Comment:
Just writing `One${round}.kt` and then referencing that in `Two.kt1 like `class Two : One${round}` should be enough to repro.

Reply:
Testing says I need to make it bigger... :/